mesh/nurbs.hpp

```diff
@@ -65,6 +65,11 @@ class KnotVector
        order @a order and number of control points @a NCP. */
    KnotVector(int order, int NCP);
 
+   /** @brief Create a KnotVector by passing in a degree, a Vector of interval
```

Comment:
This file, even before this PR, uses `degree` and `order` interchangeably, but we usually use `order` in comments. For consistency, can this be changed to `order` here? There was some discussion in PR 4153 about degree and order being different but related (degree p is order p+1 in the NURBS Book terminology). In MFEM 5.0, when we can rename functions, maybe we can choose degree or order, to stop mixing them up.

Reply:
I haven't tried to address this comment yet, since, as you mentioned it may extend beyond the scope of the current PR, which is focused on adding new constructors.  I usually avoid using the term "order" altogether, and just specifying everything in terms of polynomial degree.  It's tricky in this case, since it looks like parts of the API already use variable names `order` when they mean degree, even though some sources differ on interpretation.